Mathlib/Condensed/Discrete/Characterization.lean

```diff
@@ -0,0 +1,264 @@
+-/
+import Mathlib.Condensed.Discrete.Colimit
+import Mathlib.Condensed.Discrete.Module
+/-!
+
+# Characterizing discrete condensed sets and `R`-modules.
+
+This file proves a characterization of discrete condensed sets, discrete condensed `R`-modules over
+a ring `R`, discrete light condensed sets, and discrete light condensed `R`-modules over a ring `R`.
+see `CondensedSet.isDiscrete_tfae`, `CondensedMod.isDiscrete_tfae`, `LightCondSet.isDiscrete_tfae`,
+and `LightCondMod.isDiscrete_tfae`.
+
+Informally, we can say: The following conditions characterize a condensed set `X` as discrete
+(`CondensedSet.isDiscrete_tfae`):
+
+1. There exists a set `X'` and an isomorphism `X ≅ cst X'`, where `cst X'` denotes the constant
+   sheaf on `X'`.
+2. The counit induces an isomorphism `cst X(*) ⟶ X`.
+3. There exists a set `X'` and an isomorphism `X ≅ LocallyConstant · X'`.
+4. The counit induces an isomorphism `LocallyConstant · X(*) ⟶ X`.
+5. For every profinite set `S = limᵢSᵢ`, the canonical map `colimᵢX(Sᵢ) ⟶ X(S)` is an isomorphism.
+
+The analogues for light condensed sets, condensed `R`-modules over any ring, and light
+condensed `R`-modules are nearly identical (`CondensedMod.isDiscrete_tfae`,
+`LightCondSet.isDiscrete_tfae`, and `LightCondMod.isDiscrete_tfae`).
+-/
+
+universe u
+
+open CategoryTheory Limits Functor FintypeCat
+
+attribute [local instance] ConcreteCategory.instFunLike
+
+namespace Condensed
+
+variable {C : Type*} [Category C] [HasWeakSheafify (coherentTopology CompHaus.{u}) C]
+
+/--
+A condensed object is *discrete* if it is constant as a sheaf, i.e. isomorphic to a constant sheaf.
+-/
+abbrev IsDiscrete (X : Condensed.{u} C) := X.IsConstant (coherentTopology CompHaus)
+
+end Condensed
+
+namespace CondensedSet
+
+open CompHausLike.LocallyConstant
+
+lemma mem_locallyContant_essImage_of_isColimit_mapCocone (X : CondensedSet.{u})
+    (h : ∀ S : Profinite.{u}, IsColimit <|
+      (profiniteToCompHaus.op ⋙ X.val).mapCocone S.asLimitCone.op) :
+    X ∈ CondensedSet.LocallyConstant.functor.essImage := by
+  let e : CondensedSet.{u} ≌ Sheaf (coherentTopology Profinite) _ :=
+    (Condensed.ProfiniteCompHaus.equivalence (Type (u + 1))).symm
+  let i : (e.functor.obj X).val ≅ (e.functor.obj (LocallyConstant.functor.obj _)).val :=
+    Condensed.isoLocallyConstantOfIsColimit _ h
+  exact ⟨_, ⟨e.functor.preimageIso ((sheafToPresheaf _ _).preimageIso i.symm)⟩⟩
+
+/--
+`CondensedSet.LocallyConstant.functor` is left adjoint to the forgetful functor from condensed
+sets to sets.
+-/
+noncomputable abbrev LocallyConstant.adjunction :
+    CondensedSet.LocallyConstant.functor ⊣ Condensed.underlying (Type (u+1)) :=
+  CompHausLike.LocallyConstant.adjunction _ _
+
+open Condensed
+
+open CondensedSet.LocallyConstant List in
+theorem isDiscrete_tfae  (X : CondensedSet.{u}) :
+    TFAE
+    [ X.IsDiscrete
+    , IsIso ((Condensed.discreteUnderlyingAdj _).counit.app X)
+    , X ∈ (Condensed.discrete _).essImage
+    , X ∈ functor.essImage
```

Comment:
Since this is the "entry point" for using all these equivalent results, I suggest spelling out the full name of `functor`. Just to increase the readability of this particular statement.
Idem dito for `adjunction` below.